### ISSUE TITLE
[Bug]: React Charting Heat Map Chart tooltip appears above the wrong cell

### ISSUE THREAD

**Issue by mjiang108bl** [microsoft/fluentui#29740]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: macOS 13.6
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 2.57 GB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Browsers:
    Chrome: 119.0.6045.105
    Safari: 16.6
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://codesandbox.io/s/react-charting-heatmapchart-tooltip-wrong-cell-xw7cvv?file=/src/App.tsx

## Actual Behavior
When you hover over a cell in the heat map, a tooltip appears. It is supposed to appear above that cell. However, for certain cells, it appears above the wrong cell.

## Expected Behavior
The tooltip always appears above the correct cell.

## Notes
- It seems that if you hover over row 1 column 10 the tooltip appears over row 11 column 0, if you hover over row 1 column 11 the tooltip appears over row 11 column 1, etc. I noticed in the source code [here](https://github.com/microsoft/fluentui/blob/master/packages/react-charting/src/components/HeatMapChart/HeatMapChart.base.tsx#L315C41-L315C41) it seems to use `[row index][column index]` as an id for each cell, which is used in the onMouseOver. That means row 1 column 10 and row 11 column 0 both have an id of 110. That might be related to this bug.
- In the bug report template, it asked which library this bug affects but there was no option for @fluentui/react-charting, so I just selected a random library.

### Requested priority

Blocking

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by ankityadav4** [comment]
Thanks for reporting this issue. I have logged a task to track this issue here https://uifabric.visualstudio.com/iss/_workitems/edit/9641/

**Comment by mjiang108bl** [comment]
If the bug is caused by the code I mentioned in my bug report, I think it should be a simple fix - just change the id so it is unique for each cell, something like `[row index]-[column index]`. I could try to contribute, but it might take me some time to figure out the contributing workflow.

But I understand if it can't be prioritized right now. I currently have some time, so I will try making a contribution.

**Comment by ankityadav4** [comment]
@mjiang108bl If you are interested in making a contribution, We can help you with the steps. 
Follow https://github.com/microsoft/fluentui/wiki/Setup#basic-setup guide to get started. 
Folder structure.
The charting library is located within package/react-charting.
The source code is located in the src/ directory.

After making changes run yarn test inside react-charting to run the tests and also you will need to update the snapshots using command yarn update-snapshots . (Since the ids are changed the snapshots will fail once you run yarn test )
Let us know if you need any other information. We will help you in getting the PR merged.

**Comment by AtishayMsft** [comment]
This issue is fixed with #29800. Thanks for reporting the issue.
